real_intent/events/deep_research/__init__.py

```
@@ -0,0 +1,92 @@
+        Raises:
+            ImportError: If the OpenAI package is not installed.
+            ValueError: If the API key is invalid.
+        """
+        if not isinstance(openai_api_key, str) or not openai_api_key:
+            raise ValueError("OpenAI API key must be a truthy string.")
+
+        try:
+            from openai import OpenAI
+        except ImportError:
+            log("error", "Failed to import OpenAI. Make sure to install the package with the 'ai' extra.")
+            raise ImportError("Please install this package with the 'ai' extra.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)